code/API_definitions/Get DeviceLocation.yaml

```diff
@@ -0,0 +1,256 @@
+  termsOfService: http://swagger.io/terms/
+  contact:
+    email: project-email@sample.com
+  license:
+    name: Apache 2.0
+    url: https://www.apache.org/licenses/LICENSE-2.0.html
+  version: 0.1.0-wip
+externalDocs:
+  description: Product documentation at Camara
+  url: https://github.com/camaraproject/
+security:
+  - oAuth2ClientCredentials: []
+#  - BasicAuth: []
+#  - apiKey: []
+  - three_legged:
+    - device-location-provide
+servers:
+  - url: '{apiRoot}/{basePath}'
+    variables:
+      apiRoot:
+        default: http://localhost:9091
+        description: API root
+      basePath:
+        default: location/v0
+        description: Base path for the device location API
+paths:
+  /retrieve-location:
+    post:
+      tags:
+        - Location retrieval
+      summary: 'Execute location retrieval for a user equipment'
+      operationId: retrieveLocation
+      requestBody:
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RetrievalLocationRequest'
+        required: true
+      responses:
+        '200':
+          description: Location retrieval result
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/RetrievalLocationResponse'
+              examples: 
+                'success':
+                  summary: successful location
+                  description: The network provides device location
+                  value:
+                    latitude: 50.735851
+                    longitude: 7.10066
+                    accuracy: 2
+        '400':
+          $ref: '#/components/responses/Generic400'
+        '401':
+          $ref: '#/components/responses/Generic401'
+        '403':
+          $ref: '#/components/responses/Generic403'
+        '404':
+          $ref: '#/components/responses/Generic404'
+        '500':
+          $ref: '#/components/responses/Generic500'
+        '503':
+          $ref: '#/components/responses/Generic503'
+components:
+  securitySchemes:
+    oAuth2ClientCredentials:
+      type: oauth2
+      flows:
+        clientCredentials:
+          tokenUrl: '{tokenUrl}'
+          scopes: {}
+#    BasicAuth:
+#      type: http
+#      scheme: basic
+#    apiKey:
+#      type: apiKey
+#      description: API key to authorize requests
+#      name: apikey
+#      in: query
+    three_legged:
+      type: oauth2
+      flows:
+        authorizationCode:
+          authorizationUrl: https://auth.example.com/authorize
+          tokenUrl: https://auth.example.com/token
+          scopes:
+            device-location-provide: Provide device location
+  schemas:
+    RetrievalLocationRequest:
+      type: object
+      properties:
+        ueId:
+          $ref: '#/components/schemas/UeId'
+        uePort:
+          $ref: '#/components/schemas/Port'
+      required:
+        - ueId

```

Comment:
UeId should probably be aligned with QoD 0.9. Be renamed to Device with properties phoneNumber, networkAccessIdentifier, ipv4Address, ipv6Address with Port moving to the ipv4Address as publicPort.

Reply:
This is a fair comment. I've aligned with our existing check location API but probably we should align both device location api with Qod v0.9 if this later is enough mature.